api/specs.proto

```diff
@@ -364,3 +368,13 @@ message SecretSpec {
 	// Data is the secret payload - the maximum size is 500KB (that is, 500*1024 bytes)
 	bytes data = 2;
 }
+
+// ConfigSpec specifies user-provided configuration files.
+message ConfigSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	// Data is the config payload - the maximum size is 500KB (that is, 500*1024 bytes)
```

Comment:
Non-blocking:  is this TODO w.r.t. to multiple payloads something we still want to consider for the future?